charts/csi-baremetal-operator/templates/pre-upgrade-crds.yaml

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright © 2022 Dell Inc. or its subsidiaries.
+# All Rights Reserved.
+#
+# This software contains the intellectual property of Dell Inc.
+# or is licensed to Dell Inc. from third parties. Use of this
+# software and the intellectual property contained therein is expressly
+# limited to the terms and conditions of the License Agreement under which
+# it is provided by or on behalf of Dell Inc. or its subsidiaries.
+#
+#
+
+#
+# Create the service account for replacing CRDs during upgrade.
+#
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: "{{.Release.Namespace}}-{{.Release.Name}}-csi-baremetal-pre-crds-sa"
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    app.kubernetes.io/instance: {{.Release.Name}}
+    app.kubernetes.io/managed-by: csi-baremetal-operator
+    release: {{.Release.Name}}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
+{{- if .Values.global.registrySecret }}
+imagePullSecrets:
+  - name: {{ .Values.global.registrySecret }}
+{{- end }}
+
+#
+# Create the cluster role for replacing CRDs during upgrade.
+#
+---
+kind: ClusterRole
```

Comment:
We need to delete all roles on post upgrade hook

Reply:
Resource deletion strategy contains in the helm-based annotation
```
"helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
```

It's not need to have separate functional